### DISCUSSION TITLE
Can users connect their Metamask wallet to google account?

### DISCUSSION THREAD

**Discussion by nakulmk** [Web3Auth/Web3Auth#344] [Help]
Can users connect their Metamask wallet to google account? I want to link the users' metamask account to their web3auth google login, is that possible? Can we export the metamask private key and import into Web3Auth? Can I get a user login through metamask while using Web3Auth?

**Comment by yashovardhan** [comment]
Unfortunately not at the moment. You cannot link your metamask key with any social login through Web3Auth. This is because we cannot import keys at this point in our SDK. Our nodes follow a decentralised key generation process where you cannot import private keys into our backend server. 

However the other way is possible. The user can export their private key and import into their metamask account. 

Finally to enable a user to login using metamask, you can use our metamask adapter: https://web3auth.io/docs/sdk/web/adapters/metamask

This will enable a user to login using their metamask account, but still will not link it to their web3auth account.
